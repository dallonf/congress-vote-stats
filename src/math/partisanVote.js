export const getPartisanRating = ({ repYes, repNo, demYes, demNo }) => {
  const rep = repYes / (repYes + repNo);
  const dem = demYes / (demYes + demNo);

  if (rep > dem) {
    return { party: 'R', value: rep - dem };
  } else if (dem > rep) {
    return { party: 'D', value: dem - rep };
  } else {
    return { party: null, value: 0 };
  }
};
