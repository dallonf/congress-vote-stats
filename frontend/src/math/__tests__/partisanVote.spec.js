import { getPartisanRating } from '../partisanVote';

it('identifies a 100% Republican vote', () => {
  const data = {
    repYes: 231,
    repNo: 0,
    demYes: 0,
    demNo: 185,
  };
  expect(getPartisanRating(data)).toEqual({
    party: 'R',
    value: 1,
  });
});

it('identifies a 100% Democrat vote', () => {
  const data = {
    repYes: 0,
    repNo: 231,
    demYes: 185,
    demNo: 0,
  };
  expect(getPartisanRating(data)).toEqual({
    party: 'D',
    value: 1,
  });
});

it('identifies a unanimous vote', () => {
  const data = {
    repYes: 231,
    repNo: 0,
    demYes: 192,
    demNo: 0,
  };
  expect(getPartisanRating(data)).toEqual({
    party: null,
    value: 0,
  });
});

it('computes a (slightly) more bipartisan Republican vote', () => {
  const data = {
    demNo: 111,
    demYes: 76,
    repNo: 0,
    repYes: 231,
  };
  const rating = getPartisanRating(data);
  expect(rating.party).toEqual('R');
  expect(rating).toMatchSnapshot();
});
