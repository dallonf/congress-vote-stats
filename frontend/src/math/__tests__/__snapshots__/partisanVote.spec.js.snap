// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`computes a (slightly) more bipartisan Republican vote 1`] = `
Object {
  "party": "R",
  "value": 0.5935828877005347,
}
`;
